### DISCUSSION TITLE
Dynamic Parameters Error

### DISCUSSION THREAD

**Discussion by hkPark742** [jspsych/jsPsych#2968] [Q&A]
Hello,

I'm trying to extract the value within on_finish function and use it for a file name.

Below is the code I tried

`var survey_trial = {
            type: 'survey-text',
            preamble: "Please fill out the information below",
            questions: [
                {prompt: "Participant ID", name: 'rawID'}
            ],
            on_finish: function(data){
                var responses = JSON.parse(data.responses);
                var RawID = responses.rawID;
                jsPsych.data.addProperties({rawID:RawID});
                var fileId = RawID;
            }
        };

var fileIdd = jsPsych.data.get().last(1).values()[0].fileId;
`
The last line is giving me an error, and I could not figure out how to solve it. I was also wondering how I should change the variable "fileId" to numeric variable so that I could use it for choosing a stimuli list. Any help would be much appreciated!

**Comment by ChristopheBossens** [comment]
This is giving an error because the last line (where you use jsPsych.data.Get()) is executed immediately when you load your script. But at that point the survey trial has not been executed yet. Therefore, accessing data from the last trial will not work (because basically there is no last trial yet).

Depending on where you need to have this information, I would create a variable fileId before you create your survey trial, i.e.:

`var fileId = '';`

You can then update this variable in the on_finish function

`fileId = RawID;`

To convert a string to an integer you can use the `parseInt()` function. Alternatively, you could use the jspsych-survey-html-form plugin where you can create an input field and put a constraint on it so that it only accepts numerical values to begin with.

**Reply by hkPark742** [reply]
Thank you for your reply!

I changed my code as below

```
var CONDITION;
var fileID;

var survey_trial = {
  type: 'survey-text',
  preamble: "Please fill out the information below",
   questions: [
      {prompt: "Participant ID", name: 'rawID'}
   ],
  on_finish: function(data){
    var responses = JSON.parse(data.responses);
    var RawID = responses.rawID;
    jsPsych.data.addProperties({rawID:RawID});
    var fileId = RawID;
    var CONDITION = parseInt(RawID);
  }
};
```
and it seems that I still cannot use the fileId and CONDITION defined in on_finish function. I'm trying to  use CONDITION variable to choose stimuli list and fileId for a name of the csv file so that I can save data for each participant. My next code is

```
if (CONDITION % 2 == 0) {
  stimuli = a
} else if (CONDITION % 2 ==1) {
  stimuli = b
};
```

**Reply by ChristopheBossens** [reply]
Hi,

Could you perhaps share the full code of your experiment. Or describe what the stimuli are that you want to use later on, e.g., what are the a and b you are referring to and in what type of plugin would they occur?
